File: app/src/main/java/app/fit/fitndflow/domain/usecase/DeleteSerieUseCase.java
package app.fit.fitndflow.domain.usecase;

import android.content.Context;

import java.util.List;

import app.fit.fitndflow.data.common.SharedPrefs;
import app.fit.fitndflow.domain.common.usecase.FitUseCase;
import app.fit.fitndflow.domain.model.SerieModel;
import app.fit.fitndflow.domain.repository.FitnFlowRepository;
import io.reactivex.Single;

public class DeleteSerieUseCase extends FitUseCase<Integer, List<SerieModel>> {
    private FitnFlowRepository fitnFlowRepository;
    private Context context;

    public DeleteSerieUseCase(int inputParams, Context context, FitnFlowRepository fitnFlowRepository){
        super(inputParams);
        this.context = context;
        this.fitnFlowRepository = fitnFlowRepository;
    }
    @Override
    public Single<List<SerieModel>> buildUseCaseObservable() {
        return Single.create(emitter -> {
            try{
                String apiKey = SharedPrefs.getApikeyFromSharedPRefs(context);
                List<SerieModel> response = fitnFlowRepository.deleteSerie(inputParams, apiKey);
                emitter.onSuccess(response);
            }catch (Exception e){
                emitter.onError(e);
            }
        });
    }
}

File: app/src/main/java/app/fit/fitndflow/domain/usecase/DeleteSerieUseCase.kt
package app.fit.fitndflow.domain.usecase

import android.content.Context
import app.fit.fitndflow.data.common.SharedPrefs
import app.fit.fitndflow.domain.common.usecase.UseCase
import app.fit.fitndflow.domain.model.SerieModel
import app.fit.fitndflow.domain.repository.FitnFlowRepository
import dagger.hilt.android.qualifiers.ApplicationContext
import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.flow
import javax.inject.Inject

class DeleteSerieUseCase @Inject constructor(val fitnFlowRepository: FitnFlowRepository): UseCase<GetSerieToDeleteParams, List<SerieModel>>(){
    override fun run(params: GetSerieToDeleteParams): Flow<List<SerieModel>> = flow {
        val serieDeleted = fitnFlowRepository.deleteSerie(params.serieId)
        fitnFlowRepository.updateCurrentTrainingListCache()
        emit(serieDeleted)
    }
}

data class GetSerieToDeleteParams(val serieId: Int)